crates/router/src/db/events.rs

```diff
@@ -46,16 +46,79 @@ impl EventInterface for Store {
 impl EventInterface for MockDb {
     async fn insert_event(
         &self,
-        _event: storage::EventNew,
+        event: storage::EventNew,
     ) -> CustomResult<storage::Event, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let mut locked_events = self.event.lock().await;
+        let now = common_utils::date_time::now();
+
+        let stored_event = storage::Event {
+            #[allow(clippy::as_conversions)]
+            id: locked_events.len() as i32,
+            event_id: event.event_id,
+            event_type: event.event_type,
+            event_class: event.event_class,
+            is_webhook_notified: event.is_webhook_notified,
+            intent_reference_id: event.intent_reference_id,
+            primary_object_id: event.primary_object_id,
+            primary_object_type: event.primary_object_type,
+            created_at: now,
+        };
+
+        locked_events.push(stored_event.clone());
+
+        Ok(stored_event)
     }
     async fn update_event(
         &self,
-        _event_id: String,
-        _event: storage::EventUpdate,
+        event_id: String,
+        event: storage::EventUpdate,
     ) -> CustomResult<storage::Event, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        let mut locked_events = self.event.lock().await;
+        // find a key with the given merchant_id and key_id and update, otherwise return an error
+        let mut event_to_update = locked_events
+            .iter_mut()
+            .find(|e| e.event_id == event_id)
+            .ok_or(errors::StorageError::MockDbError)?;
+
+        match event {
+            storage::EventUpdate::UpdateWebhookNotified {
+                is_webhook_notified
+            } => {
+                event_to_update.is_webhook_notified = is_webhook_notified.is_some();
+            }
```

Comment:
```suggestion
            storage::EventUpdate::UpdateWebhookNotified {
                is_webhook_notified
            } => {
                if let Some(is_webhook_notified) = is_webhook_notified {
                    event_to_update.is_webhook_notified = is_webhook_notified;
                }
            }
```

The previous implementation would end up populating the `is_webhook_notified` field with an incorrect value.